src/common/datasource/src/object_store.rs

```diff
@@ -84,6 +85,86 @@ pub fn handle_windows_path(url: &str) -> Option<String> {
         .map(|captures| captures[0].to_string())
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum StorageType {
+    S3,
+    S3WithCache,
+    File,
+    Oss,
+    Azblob,
+    Gcs,
+}
+
+impl std::fmt::Display for StorageType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StorageType::S3 => write!(f, "S3"),
+            StorageType::S3WithCache => write!(f, "S3"),
+            StorageType::File => write!(f, "File"),
+            StorageType::Oss => write!(f, "Oss"),
+            StorageType::Azblob => write!(f, "Azblob"),
+            StorageType::Gcs => write!(f, "Gcs"),
+        }
+    }
+}
+
+impl StorageType {
+    pub fn build_storage_types_based_on_env() -> Vec<StorageType> {
+        let mut storage_types = Vec::with_capacity(4);
+        storage_types.push(StorageType::File);
+        if let Ok(bucket) = env::var("GT_S3_BUCKET") {
+            if !bucket.is_empty() {
```

Comment:
https://github.com/GreptimeTeam/greptimedb/blob/3217b56cc10eeb7f334ee08ecb72b0a0557388f5/.github/workflows/develop.yml#L178-L184

Settings for S3 reside, but it seems like bucket is empty.

Reply:
The `S3_BUCKET` is only available in this repo's secrets. A PR from a forked repo can't get this secret. So we run the test in the CI after the PR is merged.